library/library_info.ml

```diff
@@ -0,0 +1,32 @@
+(************************************************************************)
+(*         *   The Coq Proof Assistant / The Coq Development Team       *)
+(*  v      *         Copyright INRIA, CNRS and contributors             *)
+(* <O___,, * (see version control and CREDITS file for authors & dates) *)
+(*   \VV/  **************************************************************)
+(*    //   *    This file is distributed under the terms of the         *)
+(*         *     GNU Lesser General Public License Version 2.1          *)
+(*         *     (see LICENSE file for the text of the license)         *)
+(************************************************************************)
+
+(*i*)
+open Names
+(*i*)
+
+type t = Comment of string | Deprecation of Deprecation.t
+
+let info_cat = CWarnings.create_category ~name:"info" ()
+
+let warn_library_info =
+  CWarnings.create ~category:info_cat ~name:("info-library")
+  Pp.(fun (dp,s) ->
+    str "Library " ++ DirPath.print dp ++ str ":" ++ spc () ++ str s)
+
+let warn_library_deprecated =
+  (* TODO: create a warning specifically for this library?? *)
+  Deprecation.create_warning ~object_name:"Library" ~warning_name_if_no_since:"deprecated-libraries"
+    (fun dp -> DirPath.print dp)
+
+let warn_library_info ?loc (dp,infos) =
+  match infos with
+  | Deprecation t -> warn_library_deprecated ?loc (dp, t)
+  | Comment s -> warn_library_info ?loc (dp, s)
```

Comment:
Note: this file is of temporary purpose. If #18248 is merged, it will be provided by the file `information.ml` of the latter.

Reply:
I wouldn't be surprised if this is merged first (looks much simpler), so probably it should just be removed for now and added later when rebasing #18248